### PR TITLE
tyInt tyUint fit target int bit width

### PR DESCRIPTION
**Pull request by bung87**
break newly added package test see https://github.com/status-im/nim-stew/issues/151 [fixed]

### CODE REVIEW

File: compiler/types.nim [hunk 1]
```diff
@@ -797,6 +802,20 @@ proc firstFloat*(t: PType): BiggestFloat =
     internalError(newPartialConfigRef(), "invalid kind for firstFloat(" & $t.kind & ')')
     NaN
 
+proc targetSizeSignedMaxToKind*(conf: ConfigRef): TTypeKind =
+  case conf.target.intSize
+  of 8: result = tyInt64
+  of 4: result = tyInt32
+  of 2: result = tyInt32
```
Comment (PMunch) [inline compiler/types.nim]: This should be tyInt16 I believe

Comment (bung87) [reply compiler/types.nim]: yes, wrong copy paste.

File: lib/system/strs_v2.nim [hunk 2]
```diff
@@ -36,7 +36,7 @@ template frees(s) =
 
 proc resize(old: int): int {.inline.} =
   if old <= 0: result = 4
-  elif old < 65536: result = old * 2
+  elif old <= high(int16): result = old * 2
```
Comment (planetis-m) [inline lib/system/strs_v2.nim:39]: That's definitely high(uint16) == 65535

Comment (bung87) [reply lib/system/strs_v2.nim:39]: that will cause a type mismatch error, here the number doesn't need to be very specific

**Comment by Araq** [comment]
Excellent work, but why are the changes to dragonbox etc required?

**Comment by bung87** [comment]
when compile to 16 bit , can't have `1000000000'u` , it's now `uint16`

File: lib/std/private/digitsutils.nim [hunk 3]
```diff
@@ -28,13 +28,13 @@ const
 #       res.add $i
 #   doAssert res == digits100
 
-proc utoa2Digits*(buf: var openArray[char]; pos: int; digits: uint32) {.inline.} =
+proc utoa2Digits*[T: Ordinal](buf: var openArray[char]; pos: T; digits: uint32) {.inline.} =
```
Comment (Araq) [inline lib/std/private/digitsutils.nim]: Why is this change required? `pos` should be fine as `int` even if `int` is mapped to 16 bits.

Comment (bung87) [reply lib/std/private/digitsutils.nim]: I just tried https://github.com/nim-lang/Nim/commit/7456d09809dbd1b5752a0854798732d54b4763dc, in the end it still relevant to `FloatingDecimal32` which has `exponent: int32`, so to fix this will lose the possibility of string representation number larger than system bit width(by changing the input,output int type),

Comment (Araq) [reply lib/std/private/digitsutils.nim]: We don't need a string representation that supports more than the available address space.

File: lib/system/strs_v2.nim (hunk 2 above)
Comment (ringabout) [reply lib/system/strs_v2.nim:39]: Does `<=65535` work?

Comment (bung87) [reply lib/system/strs_v2.nim:39]: `int` now is `int16` when target 16bit system which is `32767`

File: lib/std/private/dragonbox.nim [hunk 4]
```diff
@@ -1219,7 +1219,7 @@ proc formatDigits*(buffer: var openArray[char]; pos: int; digits: uint64; decima
   else:
     ##  dE+123 or d.igitsE+123
     decimalDigitsPosition = 1
-  var digitsEnd = pos + int(decimalDigitsPosition + numDigits)
+  var digitsEnd:int = pos + int(decimalDigitsPosition + numDigits)
```
Comment (Araq) [inline lib/std/private/dragonbox.nim]:
```suggestion
  var digitsEnd = pos + int(decimalDigitsPosition + numDigits)
```
